static/templates/compose_control_buttons.hbs

```diff
@@ -1,15 +1,18 @@
-<input type="file" id="file_input" class="notvisible pull-left" multiple />
+{{! This file assumes message_id is only preset in message_edit_form and message_id cannot be 0.}}
+<input type="file" {{#if message_id }}id="message_edit_file_input_{{message_id}}"{{ else }}id="message_edit_file_input_{{message_id}}"{{/if}} class="notvisible pull-left" multiple />
 {{#if file_upload_enabled }}
-<a role="button" class="compose_control_button fa fa-paperclip notdisplayed" aria-label="{{t 'Attach files' }}" id="attach_files" tabindex=0 title="{{t 'Attach files' }}"></a>
+<a role="button" class="compose_control_button fa fa-paperclip notdisplayed" aria-label="{{t 'Attach files' }}" {{#if message_id }}id="attach_files_{{message_id}}"{{ else }}id="attach_files"{{/if}} tabindex=0 title="{{t 'Attach files' }}"></a>
 {{/if}}
-<a role="button" id="markdown_preview" class="compose_control_button fa fa-eye" aria-label="{{t 'Preview' }}" tabindex=0 title="{{t 'Preview' }}"></a>
-<a role="button" id="undo_markdown_preview" class="compose_control_button fa fa-edit" aria-label="{{t 'Write' }}" tabindex=0 style="display:none;" title="{{t 'Write' }}"></a>
-<a role="button" class="compose_control_button fa fa-video-camera video_link" aria-label="{{t 'Add video call' }}" tabindex=0 title="{{t 'Add video call' }}"></a>
-<a role="button" class="compose_control_button fa fa-smile-o" aria-label="{{t 'Add emoji' }}" id="emoji_map" tabindex=0 title="{{t 'Add emoji' }}"></a>
+<a role="button" {{#if message_id }}id="markdown_preview_{{message_id}}"{{ else }}id="markdown_preview"{{/if}} class="compose_control_button fa fa-eye" aria-label="{{t 'Preview' }}" tabindex=0 title="{{t 'Preview' }}"></a>
+<a role="button" {{#if message_id }}id="undo_markdown_preview_{{message_id}}"{{ else }}id="undo_markdown_preview"{{/if}} class="compose_control_button fa fa-edit" aria-label="{{t 'Write' }}" tabindex=0 style="display:none;" title="{{t 'Write' }}"></a>
+<a role="button" class="compose_control_button fa fa-video-camera video_link" aria-label="{{t 'Add video call' }}" tabindex=0 {{#if message_id }}data-message-id="{{message_id}}"{{/if}} title="{{t 'Add video call' }}"></a>
+<a role="button" class="compose_control_button fa fa-smile-o" aria-label="{{t 'Add emoji' }}" id="emoji_map" tabindex=0 {{#if message_id }}data-message-id="{{message_id}}"{{/if}} title="{{t 'Add emoji' }}"></a>
 {{#if giphy_api_available }}
 <a role="button" class="compose_control_button" aria-label="{{t 'Add GIF' }}" id="compose_box_giphy_grid" title="{{t 'Add GIF' }}">
-    <img class="compose_giphy_logo" tabindex=0 src="/static/images/GIPHY_logo.png">
+    <img class="compose_giphy_logo" tabindex=0 src="/static/images/GIPHY_logo.png" {{#if message_id }}data-message-id="{{message_id}}"{{/if}}>
```

Comment:
The clean way to do this sort of shared code is just using a class rather than an ID, and just accessing these elements as either `$("#compose").find(".markdown_preview")` or the similar one starting from the `message_edit` element.

I think given that we'll need to test everything carefully anyway, we should probably migrate to the cleaner class-based approach for these buttons rather than adding conditionals on every line like this.

(Similarly, we probably don't need a `data-message-id` on this line, if we can instead access the `zid` on the containing `.message_row` element ... which realistically should be converted to a `data-message-id` attribute in some follow-up patch)

Reply:
As a commit structure, I think we just want to replace this commit with a series of commits moving things to the class-based system (one per element), and then the final commit roughly just adds the below `drafts-link` hunk as changes to this file.

Reply:
Updated!

Reply:
Ohh! Looks like I cannot use `#zfilt{message_id}` since it is not available in `All messages` view. I don't want to use `zid` because it had very bad performance implications. Finding attributes is not a reliable way to do things too.

Reply:
I had to add `#edit_form_{message_id}` to the `form`.

Reply:
That seems like a good way to do it.